iroh-dns-server/src/lib.rs

```diff
@@ -220,8 +221,20 @@ mod tests {
     async fn integration_mainline() -> Result<()> {
         // run a mainline testnet
         let testnet = pkarr::mainline::Testnet::new(5)?;
-        let bootstrap = testnet.bootstrap.clone();
+        assert!(
+            FuturesUnordered::from_iter(
+                testnet
+                    .nodes
+                    .iter()
+                    .cloned()
+                    .map(|node| async move { node.as_async().bootstrapped().await }),
+            )
+            .any(|x| x)
```

Comment:
I was about to comment that I think this `any` here is wrong, then went back to discord to check, and yeah you chatted about this.
Perhaps worth adding a comment explaining this here?